instrumentation/google.golang.org/grpc/otelgrpc/test/grpc_stats_handler_test.go

```diff
@@ -579,3 +588,49 @@ func checkServerSpans(t *testing.T, spans []trace.ReadOnlySpan) {
 		otelgrpc.GRPCStatusCodeKey.Int64(int64(codes.OK)),
 	}, pingPong.Attributes())
 }
+
+func checkServerRecords(t *testing.T, reader metric.Reader) {
+	rm := metricdata.ResourceMetrics{}
+	err := reader.Collect(context.Background(), &rm)
+	assert.NoError(t, err)
+	require.Len(t, rm.ScopeMetrics, 1)
+	require.Len(t, rm.ScopeMetrics[0].Metrics, 5)
+	for _, m := range rm.ScopeMetrics[0].Metrics {
+		require.IsType(t, m.Data, metricdata.Histogram[int64]{})
+		data := m.Data.(metricdata.Histogram[int64])
+		for _, dpt := range data.DataPoints {
+			attr := dpt.Attributes.ToSlice()
+			method := getRPCMethod(attr)
+			assert.NotEmpty(t, method)
+			assert.ElementsMatch(t, []attribute.KeyValue{
+				semconv.RPCMethod(method),
+				semconv.RPCService("grpc.testing.TestService"),
+				otelgrpc.RPCSystemGRPC,
+				otelgrpc.GRPCStatusCodeKey.Int64(int64(codes.OK)),
+			}, attr)
+		}
+	}
```

Comment:
Can you try using https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric/metricdata/metricdatatest#AssertEqual? 

You can find example usages even in this repository.

Reply:
Ok, I will try

Reply:
You may also look at https://github.com/open-telemetry/opentelemetry-go-contrib/issues/4483

Reply:
I already change unit test by using `metricdatatest` package.

I use `AssertEqual` to compare each `metricdata.Metrics` object not `metricdata.ScopeMetrics`, since the we could not know the value of `rpc.xxx.duration` in advance. 

So for `rpc.xxx.duration`, we should use both `metricdatatest.IgnoreTimestamp()`, `metricdatatest.IgnoreValue()`. But for other metric, we could only use `metricdatatest.IgnoreTimestamp()`.

@pellared Could you help look at the code to make sure I use this package correctly?

Reply:
I simply suggest to assert against `metricdata.ScopeMetrics` and use `metricdatatest.IgnoreValue()`.

1. I do not see a lot of value in asserting the exact metrics values.
2. Checking the histograms' "values" (buckets) produces a lot of boilerplate which IMO is hard to decipher. 
3. Asserting against whole `metricdata.ScopeMetrics` makes sure that we have not forgotten to assert any metric
4. The most important things to assert are the metrics and their attributes so that we can easily spot cardinality metric attributes.

Reply:
Ok, this makes sense.

Reply:
Done